### PR TITLE
Fixes #865: Backend implementation of "detect bugs with similar domain names"

### LINKED ISSUE #865: Backend implementation of "detect bugs with similar domain names"
This issue is about coding a backend for the feature described in  #307 
This work can probably start before the design in #307 is ready. It will most likely depend on the issues database created in #788 
- [x] Use tldextract to get domain name in webhooks/helpers.py
- [x] Update db/**init**.py to add domain name field
- [x] Create new endpoint 'domainsearch' in api/endpoints.py 
- [ ] Dependency on Migration script for database. See #968

### PR DESCRIPTION
**Pull request by dshgna**
r? @hallvors @karlcow 

- [x] Use tldextract to get domain name in webhooks/helpers.py
- [x] Update db/__init__.py to add domain name field
- [x] Create new endpoint 'domainsearch' in api/endpoints.py

I successfully tested the functionality against my local test repo.

### CODE REVIEW

File: webcompat/api/endpoints.py [hunk 1]
```diff
@@ -10,10 +10,12 @@
 credentials or as a proxy on behalf of anonymous or unauthenticated users.'''
 
 import json
+import sys
```
Comment (hallvors) [inline webcompat/api/endpoints.py]: Is the sys import used for anything?

File: webcompat/db/helpers.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    '''Returns the ten most recent issues with a similar domain name'''
+    search_domain += "%"
+    session = issue_db()
+    query = (
+        session.query(WCIssue)
+        .filter(WCIssue.domain.like(search_domain))
+        .limit(10).all()
+        )
+    result_dict = []
+    for row in query:
```
Comment (hallvors) [inline webcompat/db/helpers.py]: Perhaps this code should be in a "to_dict" helper method or something?

File: webcompat/webhooks/helpers.py [hunk 3]
```diff
@@ -48,8 +49,22 @@ def set_label(label, issue_number):
     payload = [label]
     api_post('labels', payload, issue_number)
 
+def extract_domain_name(url):
+    # List of major sites where using only the domain results in losing valuable information
+    blackList = r'(\.google\.com|\.live\.com|\.yahoo\.com|go\.com|\.js$)'
```
Comment (hallvors) [inline webcompat/webhooks/helpers.py]: A simplification I was trying to suggest on IRC was something like 

```
    prefix_blacklist = ['www']
    parts = tldextract.extract(url)
    if parts[0] in prefix_blacklist:
        return '.'.join(parts[1:2])
    else:
        return '.'.join(parts[0:2])

..or something like that. Well, perhaps keep the strip('www') to handle www.mail.yahoo.com and friends ;) For bonus points, strip all entries in prefix_blacklist..

File: requirements.txt [hunk 4]
```diff
@@ -11,3 +11,4 @@ Pillow==3.1.1
 requests==2.9.1
 ua-parser==0.6.1
 WTForms==2.1
+tldextract==1.7.5
```
Comment (hallvors) [inline requirements.txt]: If tldextract is updated it presumably adds better data for the server name public prefix lists. I'd like to make the dependency >= so we get those updates.

File: webcompat/api/endpoints.py [hunk 5]
```diff
@@ -26,6 +28,8 @@
 from webcompat.helpers import normalize_api_params
 from webcompat.helpers import proxy_request
 from webcompat.issues import filter_new
+from webcompat.db import issue_db
```
Comment (hallvors) [inline webcompat/api/endpoints.py]: Is this import required?

File: webcompat/webhooks/helpers.py (hunk 3 above)
Comment (dshgna) [inline webcompat/webhooks/helpers.py]: Managed it :) Small question here on the domain blacklist: Do we need to include the \.js?

File: webcompat/api/endpoints.py (hunk 5 above)
Comment (dshgna) [inline webcompat/api/endpoints.py]: Sorry, fixed it and the other unused import.

File: webcompat/webhooks/helpers.py (hunk 3 above)
Comment (karlcow) [inline webcompat/webhooks/helpers.py]: `s/parts[0]/parts.subdomain/`
`s/parts[0:2]/parts[:]/`

but yes much better.

**Pull request closed without merging**